### PR TITLE
DesignWare drivers cleanup

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
@tbursztyka +1 for USB driver changes, could you open a own PR for that?

**Comment by tbursztyka** [comment]
@jfischer-phytec-iot Well, I think this PR is now in a ready state.

I just wonder why it's annoying me with  the dma driver, since the former name based on "cavs" was not covered by the codeowners file anyway.

File: drivers/usb/device/usb_dc_dw.c [hunk 1]
```diff
@@ -540,11 +544,110 @@ static void usb_dw_handle_enum_done(void)
 }
 
 /* USB ISR handler */
-static void usb_dw_isr_handler(void)
+static inline void usb_dw_int_rx_flvl_handler(void)
```
Comment (finikorg) [inline drivers/usb/device/usb_dc_dw.c]: On which HW have you tested the changes?

Comment (tbursztyka) [reply drivers/usb/device/usb_dc_dw.c]: none, the only board supporting it is the intel_s1000_crb and I don't have one. Could you test yourself if you happen to own that board?

Comment (finikorg) [reply drivers/usb/device/usb_dc_dw.c]: It's OK for some code clean-up but I would not change the logic too much if we cannot test it.

Comment (tbursztyka) [reply drivers/usb/device/usb_dc_dw.c]: Logic itself is 100% same, it's only some re-factorization (moving some code into own function). I think that should be fine.

File: drivers/usb/device/usb_dc_dw.c [hunk 2]
```diff
@@ -540,11 +544,110 @@ static void usb_dw_handle_enum_done(void)
...
+
+	/* Packet in RX FIFO */
+
+	ep_idx = grxstsp & USB_DW_GRXSTSR_EP_NUM_MASK;
+	status = (grxstsp & USB_DW_GRXSTSR_PKT_STS_MASK) >>
+		USB_DW_GRXSTSR_PKT_STS_OFFSET;
+	xfer_size = (grxstsp & USB_DW_GRXSTSR_PKT_CNT_MASK) >>
+		USB_DW_GRXSTSR_PKT_CNT_OFFSET;
+
+	LOG_DBG("USB OUT EP%d: RX_FLVL status %d, size %d",
```
[the remark this answers is not in the document]
Comment (tbursztyka) [reply drivers/usb/device/usb_dc_dw.c]: Indeed that was originally wrong as it's unsigned.

[the remark this answers is not in the document]
Comment (tbursztyka) [reply drivers/usb/device/usb_dc_dw.c]: however I am not changing the actual output format. it's not the purpose of that PR

File: dts/xtensa/intel/intel_s1000.dtsi [hunk 3]
```diff
@@ -180,5 +182,16 @@
 
 			status = "disabled";
 		};
+
+		usb: usb@a0000 {
+			compatible = "snps,designware-usb";
```
Comment (jfischer-phytec-iot) [inline dts/xtensa/intel/intel_s1000.dtsi]: Linux Kernel uses `snps,dwc2` for DWC2 USB controller, `compatible = "snps,dwc2";` would probably better choice.

**Review by jfischer-no — APPROVED** [review]
LGTM USB part

File: dts/xtensa/intel/intel_s1000.dtsi (hunk 3 above)
Comment (tbursztyka) [reply dts/xtensa/intel/intel_s1000.dtsi]: That would go against all other designware compatible names. Sounds like a mistake on linux side, if any.

**Comment by tbursztyka** [comment]
@nashif could this pr be applied soon? I'll need it to continue the work on DW drivers.

**Comment by tbursztyka** [comment]
CI is quite broken atm...

File: CODEOWNERS [hunk 4]
```diff
@@ -119,6 +119,7 @@
 /drivers/counter/counter_cmos.c           @andrewboie
 /drivers/display/                         @vanwinkeljan
 /drivers/display/display_framebuf.c       @andrewboie
+/drivers/dma/*dw*			  @tbursztyka
```
Comment (nashif) [inline CODEOWNERS]: spaces, no tabs.

**Review by nashif — APPROVED** [review]

**Pull request merged**